packages/checkout-extension/src/withExtension.ts

```diff
@@ -0,0 +1,7 @@
+import { createInjectHoc } from '@bigcommerce/checkout/locale';
```

Comment:
Do we need to go the HOC pattern with this? Since we are using a context we should avoid this.

Reply:
I used the analytics context's method to set up the context for extensions.
When I searched for `extends Component`, I found around 40 class component objects within `checkout-js`, including important components like `Checkout`, `Shipping`, and `Payment`.